### ISSUE TITLE
twisted.test.test_pb.NSPTestCase.tearDown calls stopListening but doesn't return the Deferred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2321]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2321|
|Type|defect|
|Created|2006-12-23 05:05:00Z|

This results in a cleanup race condition which might leave the reactor dirty.

<details><summary>Searchable metadata</summary>

```
trac-id__2321 2321
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pb pb
keywords__ 
time__1166850300000000 1166850300000000
changetime__1182673040000000 1182673040000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set status to `closed`|
|-|-|

(In [20609]) Merge pb-pager-1843+2321

Authors: ino, therve
Reviewer: exarkun
Fixes [#1843](https://github.com/twisted/twisted/issues/1843)
Fixes [#2321](https://github.com/twisted/twisted/issues/2321)

Modify FilePager so that data read is not kept in memory. In the process,
do a lot of cleanups in test_pb to match current coding standards.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
